api/v1/address/server/handler.go

```diff
@@ -0,0 +1,180 @@
+		if r.Header.Get("Content-type") == "application/json" {
+			var newAddress AddressInfo
+			reqBody, err := ioutil.ReadAll(r.Body)
+			if err == nil {
+				json.Unmarshal(reqBody, &newAddress)
+
+				PostalCode := newAddress.PostalCode
+				if len(strconv.Itoa(PostalCode)) != 6 {
+					w.WriteHeader(http.StatusUnprocessableEntity)
+					w.Write([]byte("422 - Minimum length for PostalCode is 6"))
+					return
+				}
+
+				Floor := newAddress.Floor
+				UnitNumber := newAddress.UnitNumber
+				BuildingName := newAddress.BuildingName
+				CreatedDt := time.Now().Format(time.RFC3339)
+				ModifiedDt := CreatedDt
+
+				query := fmt.Sprintf(
+					"INSERT INTO Address VALUES (NULL, %d, %d, %d, '%s', '%s', '%s')",
+					PostalCode, Floor, UnitNumber, BuildingName, CreatedDt, ModifiedDt)
+				_, err := db.Query(query)
+				if err != nil {
+					panic(err.Error())
+				}
+
+				lastId := fmt.Sprintf("SELECT LAST_INSERT_ID()")
+
+				w.WriteHeader(http.StatusCreated)
+				w.Write([]byte("201 - Address Account Number: " + lastId + " added successfully"))
+			} else {
+				w.WriteHeader(http.StatusUnprocessableEntity)
+				w.Write([]byte("422 - Please supply address information in JSON format"))
+			}
+		}
+	}
+}
+
+func updateAddress(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Content-type") == "application/json" {
+			var newAddress AddressInfo
+			reqBody, err := ioutil.ReadAll(r.Body)
+			if err == nil {
+				json.Unmarshal(reqBody, &newAddress)
+				AccountNumber := newAddress.AccountNumber
+
+				if AccountNumber < 0 {
+					w.WriteHeader(http.StatusUnprocessableEntity)
+					w.Write([]byte("422 - Please select the address you want to update"))
+					return
+				}
+
+				PostalCode := newAddress.PostalCode
+				if len(strconv.Itoa(PostalCode)) != 6 {
+					w.WriteHeader(http.StatusUnprocessableEntity)
+					w.Write([]byte("422 - Minimum length for PostalCode is 6"))
+					return
+				}
+
+				Floor := newAddress.Floor
+				UnitNumber := newAddress.UnitNumber
+				BuildingName := newAddress.BuildingName
+				ModifiedDt := time.Now().Format(time.RFC3339)
+
+				query := fmt.Sprintf(
+					"UPDATE Address SET PostalCode=%d, Floor=%d, UnitNumber=%d BuildingName='%s' ModifiedDT='%s' WHERE AccountNumber=%d",
+					PostalCode, Floor, UnitNumber, BuildingName, ModifiedDt, AccountNumber)
+				_, err := db.Query(query)
+				if err != nil {
+					panic(err.Error())
+				}
+
+				w.WriteHeader(http.StatusCreated)
+				w.Write([]byte("201 - Address Account Number: " + strconv.Itoa(AccountNumber) + " updated successfully"))
+			} else {
+				w.WriteHeader(http.StatusUnprocessableEntity)
+				w.Write([]byte("422 - Please supply address information in JSON format"))
+			}
+		}
+	}
+}
+
+func readAddress(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		params := mux.Vars(r)
+		AccountNumber := params["accountnumber"]
+		results, err := db.Query("")
```

Comment:
@shiweii I intended to use readAddress to view a single address record if params["accountnumber"] is provided, else it should query for all addresses

I wrote this initially but it's complaining about the defined variable not used in the subsequent lines.
```
if AccountNumber != "" {
  results, err := db.Query("Select * FROM Address WHERE AccountNumber='%d'", AccountNumber)
} else {
  results, err := db.Query("Select * FROM Address")
 }
 ```

So I wrote this instead, such that I can instantiate the `results` variable.. Is there a better way to do so ah?
```
results, err := db.Query("")
```

I saw this suggestion: https://stackoverflow.com/questions/21481288/variable-not-declared-inside-a-if-else-statement
```
var req *http.Request
var er error
```
If I wanna go with this approach to declare `results` variable, how to set the type?

Reply:
Maybe you create a new route to list all address instead? The other APIs uses another route for listing.

`api.Handle("/addresses", readAddresses(db)).Methods("GET")`